python_files/ml_on_iris_dataset.py

```diff
@@ -0,0 +1,192 @@
+    print(data_df.describe())
+    assert (
+        np.mean(data_df["sepal_len_cm"])
+        == data_df.describe().loc["mean", "sepal_len_cm"]
+    )
+
+    # defines X and y (class = target)
+    print_heading("Defining Covariates and Targets")
+    covariates = ["sepal_len_cm", "sepal_wid_cm", "petal_len_cm", "petal_wid_cm"]
+    iris_plants = list(data_df["class"].unique())
+    X_orig = data_df[covariates].values
+    y = data_df["class"].values
+    print(f"Columns in Covariates: {covariates}")
+    print(f"Shape of Covariates: {X_orig.shape}")
+    print(f"Targets in Class Column: {iris_plants}")
+    print(f"Shape of Class Column: {y.shape}")
+
+    # scatter plots (can also save the direct html text instead of outputting plot)
+    # same as bivariate plots
+    print_heading("Plots")
+    print("In Browser")
+    fig = px.scatter_matrix(
+        data_df,
+        dimensions=["sepal_len_cm", "sepal_wid_cm", "petal_len_cm", "petal_wid_cm"],
+        color="class",
+    )
+    fig.show()
+
+    # violin plot
+    fig = go.Figure()
+    for plant in iris_plants:
+        fig.add_trace(
+            go.Violin(
+                y=data_df["sepal_len_cm"][data_df["class"] == plant],
+                name=plant + " Sepal Length",
+                box_visible=True,
+                meanline_visible=True,
+            )
+        )
+
+        fig.add_trace(
+            go.Violin(
+                y=data_df["sepal_wid_cm"][data_df["class"] == plant],
+                name=plant + " Sepal Width",
+                box_visible=True,
+                meanline_visible=True,
+            )
+        )
+    fig.show()
+
+    # histograms plots
+    fig = px.histogram(data_df, x="petal_len_cm", color="class")
+    fig.show()
+
+    fig = px.histogram(data_df, x="petal_wid_cm", color="class")
+    fig.show()
+
+    # strip plot
+    fig = px.strip(data_df, x="petal_len_cm", y="class", orientation="h", color="class")
+    fig.show()
+
+    # parallel coordinates plot
+    # note: ['Iris-setosa': 0, 'Iris-versicolor':1, 'Iris-virginica':2]
+    plant_names, plant_indx = np.unique(data_df["class"], return_inverse=True)
+    fig = px.parallel_coordinates(
+        data_df,
+        color=plant_indx,
+        labels={
+            "color": "Plant",
+            "sepal_wid_cm": "Sepal Width",
+            "sepal_leng_cm": "Sepal Length",
+            "petal_wid_cm": "Petal Width",
+            "petal_len_cm": "Petal Length",
+        },
+        color_continuous_scale=px.colors.diverging.Tealrose,
+        color_continuous_midpoint=1,
+    )
+
+    fig.update_layout(
+        coloraxis_colorbar=dict(
+            title="Plant", tickvals=list(set(plant_indx)), ticktext=plant_names
+        )
+    )
+    fig.show()
+
+    # adds pipeline
+    print_heading("Creating Pipelines")
+    rf_pipeline = Pipeline(
+        steps=[
+            ("Normalizer", Normalizer()),
+            ("RandomForest", RandomForestClassifier(random_state=0)),
```

Comment:
Does it matter if all of the random state are set to zero? Do they need to be different between models?

Reply:
You can choose any number. It can be 93784683649834. As long as you use the same number, you will always get the same performance or output. That's why I included it. So the user can see the same results.

Reply:
No you can use any number actually. I was just setting the seed to 0 (first number I thought of) so that the users can reproduce my results.